### PR TITLE
Script for defect 2458

### PR DESCRIPTION
**Pull request by TMyza**
ATF Test Scripts to check [#2458](https://github.com/smartdevicelink/sdl_core/issues/2458)

Summary
A set of test scripts to verify that issue is fixed

This PR is **not ready** for review.
This is **draft** version of the script.

### CODE REVIEW

File: test_scripts/Defects/5_0/2458_Need_change_OnButtonEventNotification_logic_for_OK_button.lua [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+local common = require('user_modules/sequences/actions')
+local runner = require('user_modules/script_runner')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local Functions ]]
+local function hmiLeveltoLimited(pAppId)
+    common.getHMIConnection(pAppId):SendNotification("BasicCommunication.OnAppDeactivated",
+        { appID = common.getHMIAppId(pAppId) })
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2458_Need_change_OnButtonEventNotification_logic_for_OK_button.lua]: please use the same alignment through  all script.
Check lines 25, 27 , 40, 49 and so on.

File: test_scripts/Defects/5_0/2458_Need_change_OnButtonEventNotification_logic_for_OK_button.lua [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+runner.Step("SubscribeButton App1", subscribeButton, { 1 })
+
+runner.Step("Register App2", common.registerAppWOPTU, { 2 })
+runner.Step("Activate App2(HMI Level FULL)", common.activateApp, { 2 })
+runner.Step("SubscribeButton App2", subscribeButton, { 2 })
+
+-- [[ Test ]]
+runner.Title("Test")
+runner.Step("OnButtonEvent without App ID", OnButtonEventWithoutAppID)
+runner.Step("OnButtonEvent with App ID", OnButtonEventWithAppID, { 1, 2 })
```
Comment (aderiabin) [inline test_scripts/Defects/5_0/2458_Need_change_OnButtonEventNotification_logic_for_OK_button.lua]: @TMyza Function params can be removed and used internaly values 1 and 2

**Comment by TMyza** [comment]
@GetmanetsIrina Please review added changes according to comments for PR

**Comment by TMyza** [comment]
@GetmanetsIrina Please review added changes according to comments for PR

**Comment by TMyza** [comment]
@GetmanetsIrina Please review added changes according to comments for PR

**Comment by TMyza** [comment]
@GetmanetsIrina Please review added changes according to comments for PR

File: test_scripts/Defects/5_0/2458_Need_change_OnButtonEventNotification_logic_for_OK_button.lua (hunk 2 above)
Comment (TMyza) [reply test_scripts/Defects/5_0/2458_Need_change_OnButtonEventNotification_logic_for_OK_button.lua]: Done

Comment (TMyza) [reply test_scripts/Defects/5_0/2458_Need_change_OnButtonEventNotification_logic_for_OK_button.lua]: Done

Comment (TMyza) [reply test_scripts/Defects/5_0/2458_Need_change_OnButtonEventNotification_logic_for_OK_button.lua]: Done

**Comment by dboltovskyi** [comment]
Livio team please do not review this PR for now since it hasn't passed our internal review yet.
Review and appropriate updates going be done till 1st of November.
Thank you.

**Comment by dboltovskyi** [comment]
The internal review iteration has been moved to the LuxoftSDL.

**Pull request closed without merging**